### ISSUE TITLE
vpoller hangs if connection to vSphere is lost

### ISSUE THREAD

**Issue by sirmaxru** [dnaeon/py-vpoller#212]
if connection to vSphere downs, vPoller hangs or smth like this
the only way to fix it is to kill -9 all workers, then restart proxy and workers
I'm using vpoller with zabbix
I will try to do future investgations.

**Comment by Emmanux** [comment]
Same here. It's  damning.

**Comment by rockaut** [comment]
I know that issue from other pyvmomi usages. Same problem there i think.
But isn't that more a problem of https://github.com/dnaeon/py-vconnector ?

**Comment by dnaeon** [comment]
@rockaut correct, connection is handled by `py-vconnector`, but we should have a mechanism to remove stale connections. I haven't been paying attention to vPoller for a while, so I haven't had a chance to check this out.

**Comment by rapgro** [comment]
Somehow garbage collection initiated from a monitoring system, or alike.
